apps/web-giddh/src/app/shared/primary-sidebar/company-branch/company-branch.component.ts

```diff
@@ -0,0 +1,429 @@
+        if(design === 2) {
+            this.viewingCompany = company;
+        }
+        if (!company.branches || reloadBranches) {
+            company.branches = [];
+            this.branchRefreshInProcess = true;
+            let branchFilterRequest: BranchFilterRequest = { from: '', to: '', companyUniqueName: company.uniqueName };
+            this.settingsBranchService.GetAllBranches(branchFilterRequest).subscribe(response => {
+                if (response?.status === "success") {
+                    this.branchList = response.body.sort(this.generalService.sortBranches);
+                    company.branches = this.branchList;
+                    this.companyBranches = company;
+                    this.companyBranches.branchCount = this.branchList?.length;
+                    this.branchRefreshInProcess = false;
+
+                    if (this.searchBranch) {
+                        this.filterBranchList(this.searchBranch);
+                    }
+
+                    this.changeDetectorRef.detectChanges();
+                } else {
+                    this.branchList = [];
+                    company.branches = this.branchList;
+                    this.companyBranches = company;
+                    this.companyBranches.branchCount = this.branchList?.length;
+                    this.branchRefreshInProcess = false;
+
+                    this.changeDetectorRef.detectChanges();
+                }
+            });
+        } else {
+            this.branchList = company.branches;
+            this.companyBranches = company;
+        }
+    }
+
+    /**
+     * This will show all branches of company in branch tab
+     *
+     * @memberof CompanyBranchComponent
+     */
+    public showAllBranches(company: any): void {
+        if(company?.branchCount > 1) {
+            setTimeout(() => {
+                if (this.staticTabs && this.staticTabs.tabs[1]) {
+                    this.staticTabs.tabs[1].active = true;
+                }
+            }, 20);
+        } else {
+            if(company?.uniqueName !== this.activeCompany?.uniqueName) {
+                this.changeCompany(company?.uniqueName, '', false);
+            }
+        }
+    }
+
+    /**
+     * Callback for select tab event
+     *
+     * @param {string} tabName
+     * @memberof CompanyBranchComponent
+     */
+    public tabChanged(tabName: string): void {
+        this.activeTab = tabName;
+        this.searchBranch = "";
+        this.filterBranchList(this.searchBranch);
+
+        if(tabName === "company") {
+            this.companyBranches = this.activeCompany;
+            this.companyBranches.branches = this.currentCompanyBranches;
+            this.branchList = this.currentCompanyBranches;
+            this.changeDetectorRef.detectChanges();
+        }
+    }
+
+    /**
+     * This will filter branches
+     *
+     * @param {*} event
+     * @memberof CompanyBranchComponent
+     */
+    public filterBranchList(event: any): void {
```

Comment:
Please check a similar method `filterBranch` is also present in `primary-sidebar.component.ts`, do remove it from there if unused.